### PR TITLE
Generate cassandra cql script

### PR DESCRIPTION
**Pull request by eladBenedict**

### CODE REVIEW

File: kanalony-storage-access/src/main/scala/kanalony/storage/generator/CassandraCqlGenerator.scala [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+/**
+ * Created by elad.benedict on 3/27/2016.
+ */
+object CassandraCqlGenerator {
+
+  def main (args: Array[String]) {
+
+    val keySpaceName = "kanalony_user_activity"
+    var res = List(s"CREATE KEYSPACE IF NOT EXISTS ${keySpaceName} WITH replication = {'class': 'SimpleStrategy', 'replication_factor': '1'}  AND durable_writes = true;")
```
Comment (ofirski) [inline kanalony-storage-access/src/main/scala/kanalony/storage/generator/CassandraCqlGenerator.scala]: I don't think we should include the keyspace creation here, as it has different replication factors on different deployments (local = 1, staging = X, production = Y)

File: kanalony-storage-access/src/main/scala/kanalony/storage/generator/CassandraCqlGenerator.scala [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    val tableNamePlaceholder = "%TABLE_NAME%"
+    val columnDefsPlaceholder = "%COLUMN_DEFS%"
+    val partitionKeyColsPlaceholder = "%PARTITION_KEY_COLS%"
+    val clusteringColsPlaceholder = "%CLUSTERING_KEY_COLS%"
+    val timeColPlaceholder = "%TIME_COL%"
+
+    val tableCreationTemplate = s"""CREATE TABLE IF NOT EXISTS ${keySpaceName}.%TABLE_NAME% (
+                                  |    %COLUMN_DEFS%
+                                  |    PRIMARY KEY ((%PARTITION_KEY_COLS%), %CLUSTERING_KEY_COLS%)
+                                  |) WITH CLUSTERING ORDER BY (%TIME_COL% DESC);"""
```
Comment (ofirski) [inline kanalony-storage-access/src/main/scala/kanalony/storage/generator/CassandraCqlGenerator.scala]: Can you please add 2 optional params to the TablesMetadata properties:
default_time_to_live: default=0   (on minutely tables should be 15552000 which is 180 days; on 10sec tables should be 604800 which is 7 days)
gc_grace_seconds: default=43200 (12 hours)

cc: @olampert

File: kanalony-storage-access/src/main/scala/kanalony/storage/generator/CassandraCqlGenerator.scala [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+      case ColumnType.DateTime => "timestamp"
+      case ColumnType.Int => "int"
+      case ColumnType.Long => "bigint"
+      case ColumnType.String => "text"
+    }
+
+    TablesMetadata.metadata.map(tm => {
+        val tableScript = tableCreationTemplate
+          .replace(tableNamePlaceholder, tm.tableName)
+          .replace(columnDefsPlaceholder, tm.columns.map(c => s"${c.name} ${getCassandraType(c.typeName)}").mkString("", ",\n", ",\n"))
```
Comment (ofirski) [inline kanalony-storage-access/src/main/scala/kanalony/storage/generator/CassandraCqlGenerator.scala]: to improve indentation maybe we can change the mkString to: `mkString("", ",\t\n", ",\t\n")`

File: kanalony-storage-access/src/main/scala/kanalony/storage/generator/CassandraCqlGenerator.scala (hunk 1 above)
Comment (eladBenedict) [inline kanalony-storage-access/src/main/scala/kanalony/storage/generator/CassandraCqlGenerator.scala]: Where would you like it? I can put it in a separate file. Currently (prior to this commit) it's part of the table creation script

Comment (ofirski) [inline kanalony-storage-access/src/main/scala/kanalony/storage/generator/CassandraCqlGenerator.scala]: A separate file seems like a good idea.

File: kanalony-storage-access/src/main/scala/kanalony/storage/generator/CassandraCqlGenerator.scala (hunk 3 above)
Comment (eladBenedict) [inline kanalony-storage-access/src/main/scala/kanalony/storage/generator/CassandraCqlGenerator.scala]: Updated to mkString("", ",\n\t", ",")

File: kanalony-storage-access/src/main/scala/kanalony/storage/generator/CassandraCqlGenerator.scala (hunk 2 above)
Comment (eladBenedict) [inline kanalony-storage-access/src/main/scala/kanalony/storage/generator/CassandraCqlGenerator.scala]: Done

File: kanalony-storage-access/src/main/scala/kanalony/storage/generator/CassandraCqlGenerator.scala (hunk 1 above)
Comment (eladBenedict) [inline kanalony-storage-access/src/main/scala/kanalony/storage/generator/CassandraCqlGenerator.scala]: Done

**Pull request merged**